lib/sles4sap/sap_deployment_automation_framework/basetest.pm

```diff
@@ -125,6 +128,66 @@ sub post_fail_hook {
         }
     }
 
+    # Update logs (except deployment VM logs) before cleanup
+    if (get_required_var('TEST') !~ /_deploy_/) {
+        # Upload logs appearing in deployer VM
+        record_info('Upload logs appearing in deloyer VM');
+        # Prepare deployer logs path
+        my $sap_sid = get_required_var('SAP_SID');
+        my $config_root_path = get_sdaf_config_path(
+            deployment_type => 'sap_system',
+            vnet_code => get_workload_vnet_code(),
+            env_code => get_required_var('SDAF_ENV_CODE'),
+            sdaf_region_code => convert_region_to_short(get_required_var('PUBLIC_CLOUD_REGION')),
+            sap_sid => $sap_sid);
+        my $logs_dir = $config_root_path . '/logs/';
+        connect_target_to_serial();
```

Comment:
Could it fails? What is going to happens of other part of the post_fail_hook if it fails here?